pyproject.toml

```diff
@@ -24,12 +24,16 @@ python = ">=3.9,<3.13"
 qibo = "^0.2.11"
 qibo_client = "^0.0.9"
 qiskit_ibm_runtime = ">=0.17"
-qiskit_ibm_provider = ">=0.8.0"
+qiskit_braket_provider = ">=0.4.1"
+qiskit = "^1.1.0"
```

Comment:
do we need explicit dependence from `qiskit` at this stage, namely if we stick with the current `qibo -> braket` conversion layer implemented?

Reply:
Hmm. so I was thinking that we wait till our next discussion to decide the next step forward, whether we use the current `qibo -> braket` version or go `qibo -> qasm -> qiskit -> braket`. 

For`qibo -> braket`, we don't need qiskit dependencies. I'll clean up qiskit dependencies and we can go ahead with merging for now. Then I'll open an issue as you mentioned [above](https://github.com/qiboteam/qibo-cloud-backends/pull/30#pullrequestreview-2332862922) about the `qibo -> qasm -> qiskit -> braket`.  

For `qibo -> qasm -> qiskit -> braket`, we'll leave the qiskit dependencies as is, but we need to take note the issue about the Phase RX gate.

These are my thoughts, just so we're on the same page.  :-)

Reply:
For the moment I think we can stick with the current `qibo -> braket` conversion, thus I would remove the `qiskit` dependency. We will reassess this once we come up with a final decision for that.

Reply:
Sounds good, `qibo -> braket` it is for now. Thanks @BrunoLiegiBastonLiegi !

Reply:
Sorry, quick question. If I update the `pyproject.toml` file, do I need to update the `poetry.lock` file using `poetry lock` and then pushing the updated `pyproject.toml` and `poetry.lock` files? Or will updating `pyproject.toml` file suffice?

Reply:
You should update the lock as well I believe, because in this case, for instance you removed a dependency that brakes the tests (I mean they were failing regardless but let's pretend that's not the case), and this is kept track of from the lock. Therefore, if you just restore the dependency in the `pyproject` without regenerating the lock the environment in the CI will still be built on top of the old lock, thus with the missing dependency.

Reply:
@mho291 `pyproject.toml` is the specification you're giving (the input to the `poetry lock` program), while `poetry.lock` is the generated snapshot (program's output).

If it were deterministic (given one input you'd always obtain the same output), we would have never stored the `poetry.lock` file, since it could be regenerated (with an extra computation of a few seconds...).
The motivation to store the lock file is that it depends on the time you're generating it, since it may change if new packages are getting published/yanked. In principle, given all timestamps you may be able to make the generation process deterministic, and thus you would only need to store `pyproject.toml` and a timestamp (already part of the commit). But it's not simple, probably expensive, and sometimes impossible. That's why we need the lock file (in summary: for reproducibility).

Reply:
Thank you @BrunoLiegiBastonLiegi and @alecandido !